driver-core/src/main/com/mongodb/internal/connection/MessageSettings.java

```diff
@@ -27,8 +27,20 @@
  */
 @Immutable
 public final class MessageSettings {
+    /**
+     * <a href="https://www.mongodb.com/docs/current/reference/command/hello/#mongodb-data-hello.maxBsonObjectSize">
+     * {@code maxBsonObjectSize}</a>.
+     */
     private static final int DEFAULT_MAX_DOCUMENT_SIZE = 0x1000000;  // 16MB
+    /**
+     * <a href="https://www.mongodb.com/docs/current/reference/command/hello/#mongodb-data-hello.maxMessageSizeBytes">
+     * {@code maxMessageSizeBytes}</a>.
+     */
     private static final int DEFAULT_MAX_MESSAGE_SIZE = 0x2000000;   // 32MB
+    /**
+     * <a href="https://www.mongodb.com/docs/current/reference/command/hello/#mongodb-data-hello.maxWriteBatchSize">
+     * {@code maxWriteBatchSize}</a>.
+     */
     private static final int DEFAULT_MAX_BATCH_COUNT = 1000;
```

Comment:
Is my understanding (links to the docs) correct here? I am specifically not sure about `DEFAULT_MAX_BATCH_COUNT`: we use 1000, while the MongoDB docs for [`maxWriteBatchSize`](https://www.mongodb.com/docs/current/reference/command/hello/#mongodb-data-hello.maxWriteBatchSize) says `100_000`.

Reply:
MongoDB 3.4 and previous had 1,000. In 3.6 it went up to 100,000: https://www.mongodb.com/docs/v4.0/reference/command/isMaster/#isMaster.maxWriteBatchSize

Reply:
👍

Reply:
Right, this is just the driver default, if `hello` doesn't provide one.  In practice it always does now.